### DISCUSSION TITLE
Code-Style - Clang-Format Styles

### DISCUSSION THREAD

**Discussion by kykrueger** [nebulastream/nebulastream#14] [Polls]
In the Nebula-Stream project, there is documentation suggesting the use of the [google style guide for c++](https://google.github.io/styleguide/cppguide.html). However, in practice, a [modified](https://github.com/nebulastream/nebulastream-public/blob/master/.clang-format) [LLVM style guide](https://llvm.org/docs/CodingStandards.html) has been enforced in the project with a .clang-format file and CI that converts to it.

Another suggestion recieved has been to look at [ClickHouse's Style Guide](https://clickhouse.com/docs/en/development/style)

Example diff for pure LLVM: https://github.com/nebulastream/nebulastream-public/pull/15
Example diff for pure google: https://github.com/nebulastream/nebulastream-public/pull/16
Example diff for clickhouse: https://github.com/nebulastream/nebulastream-public/pull/17

**Comment by alepping** [comment]
Maybe not everyone is aware, but the Google style guide needs to apply to all of Google's c++ projects, including decades old projects. Therefore a line break at 80 characters is also in the google c++ guidelines. Whether it really reflects modern c++ is also debatable.

**Reply by kykrueger** [reply]
Good point, I'm a bit of a fan of the classic 90 character lines, but 90 or 100 also fits well enough on one half of a laptop display for me.
